src/Bridge/Doctrine/EventListener/PublishMercureUpdatesListener.php

```diff
@@ -0,0 +1,175 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Bridge\Doctrine\EventListener;
+
+use ApiPlatform\Core\Api\IriConverterInterface;
+use ApiPlatform\Core\Api\ResourceClassResolverInterface;
+use ApiPlatform\Core\Api\UrlGeneratorInterface;
+use ApiPlatform\Core\Exception\InvalidArgumentException;
+use ApiPlatform\Core\Exception\RuntimeException;
+use ApiPlatform\Core\Metadata\Resource\Factory\ResourceMetadataFactoryInterface;
+use ApiPlatform\Core\Util\ClassInfoTrait;
+use Doctrine\ORM\Event\OnFlushEventArgs;
+use Symfony\Component\ExpressionLanguage\ExpressionLanguage;
+use Symfony\Component\Mercure\Update;
+use Symfony\Component\Messenger\MessageBusInterface;
+use Symfony\Component\Serializer\SerializerInterface;
+
+/**
+ * Publish resources updates to the Mercure hub.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ *
+ * @experimental
+ */
+final class PublishMercureUpdatesListener
+{
+    use ClassInfoTrait;
+
+    private $resourceClassResolver;
+    private $iriConverter;
+    private $resourceMetadataFactory;
+    private $serializer;
+    private $messageBus;
+    private $publisher;
+    private $expressionLanguage;
+    private $createdEntities;
+    private $updatedEntities;
+    private $deletedEntities;
+
+    public function __construct(ResourceClassResolverInterface $resourceClassResolver, IriConverterInterface $iriConverter, ResourceMetadataFactoryInterface $resourceMetadataFactory, SerializerInterface $serializer, MessageBusInterface $messageBus = null, callable $publisher = null, ExpressionLanguage $expressionLanguage = null)
+    {
+        if (null === $messageBus && null === $publisher) {
+            throw new InvalidArgumentException('A message bus or a publisher must be provided.');
+        }
+
+        $this->resourceClassResolver = $resourceClassResolver;
+        $this->iriConverter = $iriConverter;
+        $this->resourceMetadataFactory = $resourceMetadataFactory;
+        $this->serializer = $serializer;
+        $this->messageBus = $messageBus;
+        $this->publisher = $publisher;
+        $this->expressionLanguage = $expressionLanguage ?? class_exists(ExpressionLanguage::class) ? new ExpressionLanguage() : null;
+        $this->reset();
+    }
+
+    /**
+     * Collects created, updated and deleted entities.
+     */
+    public function onFlush(OnFlushEventArgs $eventArgs)
+    {
+        $uow = $eventArgs->getEntityManager()->getUnitOfWork();
+
+        foreach ($uow->getScheduledEntityInsertions() as $entity) {
+            $this->storeEntityToPublish($entity, 'createdEntities');
+        }
+
+        foreach ($uow->getScheduledEntityUpdates() as $entity) {
```

Comment:
Hum, good question. Is it really necessary because we don't do that in https://github.com/api-platform/core/blob/master/src/Bridge/Doctrine/EventListener/PurgeHttpCacheListener.php either

Reply:
I'll wrote a test it and let you kown

Reply:
Commit it please! Extra tests are always good to have!